python/setup.py

```diff
@@ -527,7 +527,7 @@ def build(build_python, build_java, build_cpp):
     # version of Python to build packages inside the build.sh script. Note
     # that certain flags will not be passed along such as --user or sudo.
     # TODO(rkn): Fix this.
-    if not os.getenv("SKIP_THIRDPARTY_INSTALL"):
+    if not os.getenv("SKIP_THIRDPARTY_INSTALL_CONDA_FORGE"):
```

Comment:
This is a way we vendor those libraries. If we skip them for conda forge, does that mean conda installed Ray won't have these libraries and people need to install them manually?

Reply:
TL;DR: People do not need to install them manually. There are runtime dependencies that are installed when ray is installed. For the purposes of the conda build, the "new" aiohttp stanza is equivalent to the other third-party installs and should be toggled behind the environment varaible.

And in too much detail:

General runtime dependencies are specified elsewhere (for pip installations, they appear [elsewhere in this file](https://github.com/ray-project/ray/blob/53813e5ef59960b1dbf93f5e61417283c2be1a85/python/setup.py#L251), for conda-forge they are part of [the recipe](https://github.com/conda-forge/ray-packages-feedstock/blob/9a389981ff37da2ac88d8751b6b26b941e64f170/recipe/meta.yaml#L90)). This stanza was part of PR #37158, with the comment
> This PR introduces a new standalone runtime_env_agent that is a HTTP server and is independent from dashboard_agent. It uses aiohttp so we vendor it in wheel building time.

The PR was motivated by #35472, so I assume the issue was to eliminate problems due to different version of grpc/grpcio being used within the same ray environment.

Perhaps @rynewang can explain it better, but I think the intention is to vendor all the requirements for a standalone python program and to run it independently of whatever versions of packages pip might install for the ray environment. For instance, if a user installs a package after installing ray, that may pull in a non-compliant version of grpcio or protocolbuf, and ray will not work. 

I am not sure that PR attained its goal, since the vendored libraries include only the same dependencies as are in the requirements for the default ray runtime, and as far as I can tell grpc itself is not being vendored. For instance, in my linux installation of pip-downloaded ray 2.35.0 I see:
- in `site-packages/ray/_private/runtime_env/agent/thirdparty_files`: aiohappyeyeballs-2.4.0, aiohttp-3.10.5, aiosignal-1.3.1, attrs-24.2.0, frozenlist-1.4.1, idna-3.7, multidict-6.0.5, yarl-1.9.4
- in `site-packages/ray/thirdparty_files`: colorama-0.4.6, psutil-6.0.0, setproctitle-1.2.2.

As far as I can tell, none of those contain a grpc runtime, so there is no harm in using the conda-controlled runtime dependencies. Note that conda is slightly better about building all its packages with the same versions of grpc than PyPI. This is due to the way packages are built for the two distribution systems. Conda builds all the packages on its own systems and synchronizes the version of grpc used by all of them, where PyPI allows anyone to upload binary wheels built with whatever dependencies they decide are valid.